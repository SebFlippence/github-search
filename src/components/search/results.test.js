import React from 'react';
import { render } from '@testing-library/react';

import SearchResults from './results';

test('renders correctly filtering out any users without ids', () => {
  const { baseElement, getByText } = render(
    <SearchResults
      results={[
        {
          id: 'my-id',
          login: 'JohnDoe',
          name: 'John Doe',
          email: 'john.doe@example.com',
          avatarUrl: 'https://example.com/foo.png'
        },
        {
          login: 'Bad data',
          name: '',
          email: '',
          avatarUrl: ''
        },
        {
          id: 'another-user',
          login: 'AnotherJohnDoe',
          name: 'Another John Doe',
          email: 'another.john.doe@example.com',
          avatarUrl: 'https://example.com/foo.other.png'
        }
      ]}
    />
  );

  const searchResultsCountElement = getByText('Search results (2)');
  expect(searchResultsCountElement).toBeInTheDocument();

  expect(baseElement).toMatchSnapshot();
});
