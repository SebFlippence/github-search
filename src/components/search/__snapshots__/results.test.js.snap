// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly filtering out any users without ids 1`] = `
<body>
  <div>
    <div
      class="w-full max-w-md px-12 py-8"
    >
      <h2
        class="text-xl block font-bold pb-2 py-4"
      >
        Search results (
        2
        )
      </h2>
      <ul>
        <li
          class="py-3"
        >
          <button
            class="text-left flex items-center m-2"
            type="button"
          >
            <img
              alt="Avatar of JohnDoe"
              class="w-10 h-10 rounded-full mr-4"
              src="https://example.com/foo.png"
            />
            <div>
              <p
                class="text-gray-900 leading-none underline"
              >
                JohnDoe
              </p>
              <p
                class="text-gray-600"
              >
                John Doe
                 
                <span
                  class="italic"
                >
                  <a
                    class="hover:underline"
                    href="mailto:john.doe@example.com"
                  >
                    john.doe@example.com
                  </a>
                </span>
              </p>
            </div>
          </button>
        </li>
        <li
          class="py-3"
        >
          <button
            class="text-left flex items-center m-2"
            type="button"
          >
            <img
              alt="Avatar of AnotherJohnDoe"
              class="w-10 h-10 rounded-full mr-4"
              src="https://example.com/foo.other.png"
            />
            <div>
              <p
                class="text-gray-900 leading-none underline"
              >
                AnotherJohnDoe
              </p>
              <p
                class="text-gray-600"
              >
                Another John Doe
                 
                <span
                  class="italic"
                >
                  <a
                    class="hover:underline"
                    href="mailto:another.john.doe@example.com"
                  >
                    another.john.doe@example.com
                  </a>
                </span>
              </p>
            </div>
          </button>
        </li>
      </ul>
    </div>
  </div>
</body>
`;
